tools/mot_error_visualization.py

```diff
@@ -0,0 +1,266 @@
+        assert len(bbox_colors) == 3
+    else:
+        bbox_colors = sns.color_palette(n_colors=3)
+        bbox_colors = [[int(255 * _c) for _c in bbox_color][::-1]
+                       for bbox_color in bbox_colors]
+    if isinstance(img, str):
+        img = mmcv.imread(img)
+
+    img_shape = img.shape
+    bboxes[:, 0::2] = np.clip(bboxes[:, 0::2], 0, img_shape[1])
+    bboxes[:, 1::2] = np.clip(bboxes[:, 1::2], 0, img_shape[0])
+
+    for bbox, wrong_type, id in zip(bboxes, wrong_types, ids):
+        x1, y1, x2, y2 = bbox[:4].astype(np.int32)
+        score = float(bbox[-1])
+
+        # bbox
+        bbox_color = bbox_colors[wrong_type]
+        cv2.rectangle(img, (x1, y1), (x2, y2), bbox_color, thickness=thickness)
+
+        # FN does not have id and score
+        if wrong_type == 1:
+            continue
+
+        # id
+        text = str(id)
+        width = len(text) * text_width
+        img[y1 + text_height:y1 + text_height * 2,
+            x1:x1 + width, :] = bbox_color
+        cv2.putText(
+            img,
+            str(id), (x1, y1 + text_height * 2 - 2),
+            cv2.FONT_HERSHEY_COMPLEX,
+            font_scale,
+            color=(0, 0, 0))
+
+        # score
+        text = '{:.02f}'.format(score)
+        width = (len(text) - 1) * text_width
+        img[y1:y1 + text_height, x1:x1 + width, :] = bbox_color
+        cv2.putText(
+            img,
+            text, (x1, y1 + text_height - 2),
+            cv2.FONT_HERSHEY_COMPLEX,
+            font_scale,
+            color=(0, 0, 0))
+
+    if show:
+        mmcv.imshow(img, wait_time=wait_time)
+    if out_file is not None:
+        mmcv.imwrite(img, out_file)
+
+    return img
+
+
+def main():
+    args = parse_args()
+
+    assert args.show or args.out_dir, \
+        ('Please specify at least one operation (show the results '
+         '/ save the results) with the argument "--show" or "--out-dir"')
+
+    if args.out_dir:
+        assert args.out_image or args.out_video, \
+            ('Please specify at least one type (save as images / save '
+             'as videos) with the argument "--out-image" or "--out-video"')
+
+    if not args.result_file.endswith(('.pkl', 'pickle')):
+        raise ValueError('The result file must be a pkl file.')
+
+    if args.out_dir is not None:
+        os.makedirs(args.out_dir, exist_ok=True)
+
+    print('This script visualizes the error for multiple object tracking. '
+          'By Default, the yellow bounding box denotes false negative, '
+          'the blue bounding box denotes the false positive '
+          'and the green bounding box denotes ID switch.')
+
+    cfg = Config.fromfile(args.config)
+    dataset = build_dataset(cfg.data.val, dict(test_mode=True))
+    results = mmcv.load(args.result_file)
+
+    # format the results to txts
+    resfiles, names, tmp_dir = dataset.format_results(results, None, ['track'])
+
+    for name in names:
+        print(f'Start processing video {name}')
```

Comment:
This part is specific to the MOT Challenge dataset. We will support more MOT datasets in the near future, and different dataset has different data processes. Consider package them as a function or use other methods.